File: go/otp_authenticator_test.go
/**
 * IMPORTANT: This test uses Mailosaur Authenticator, which may not
 * be enabled on your account yet (see https://mailosaur.com/app/devices)
 */

package mailosaurExample

import (
	"fmt"
	"os"
	"testing"

	"github.com/joho/godotenv"
	"github.com/mailosaur/mailosaur-go"
)

func TestOtpAuthenticator(t *testing.T) {
  t.Skip("reason")

  godotenv.Load()
  
  apiKey := os.Getenv("MAILOSAUR_API_KEY")

  // Instantiate Mailosaur client with api key
  m := mailosaur.New(apiKey)

	/**
   * This is a base32-encoded shared secret.
   * Typically this is the value shown to a user if they cannot scan an on-screen QR code.
   * Learn more at https://mailosaur.com/docs/mfa
  */
  sharedSecret := "ONSWG4TFOQYTEMY="
	currentOtp, err := m.Devices.Otp(sharedSecret)

	if (err != nil) {
    t.Error(err)
  }

  fmt.Printf("\nAuthenticator otp code - " + currentOtp.Code + "\n") // "564214"
}

File: selenium/java/src/test/java/demo/OtpAuthenticatorTest.java
/**
 * IMPORTANT: This test uses Mailosaur Authenticator, which may not
 * be enabled on your account yet (see https://mailosaur.com/app/devices)
 */

package demo;

import org.junit.Ignore;
import org.junit.Test;
import io.github.cdimascio.dotenv.Dotenv;
import com.mailosaur.MailosaurClient;
import com.mailosaur.MailosaurException;
import com.mailosaur.models.*;
import java.io.IOException;

public class OtpAuthenticatorTest {
  @Ignore("Reason")
  @Test public void retrieveOneTimePasscode() throws IOException, MailosaurException {
    Dotenv dotenv = Dotenv.load();

    String apiKey = dotenv.get("mailosaurApiKey");

    // Instantiate Mailosaur client with api key
    MailosaurClient mailosaur = new MailosaurClient(apiKey);

    /**
     * This is a base32-encoded shared secret.
     * Typically this is the value shown to a user if they cannot scan an on-screen QR code.
     * Learn more at https://mailosaur.com/docs/mfa
    */
    String sharedSecret = "ONSWG4TFOQYTEMY=";
    OtpResult currentOtp = mailosaur.devices().otp(sharedSecret);

    System.out.printf("\nAuthenticator otp code -" + currentOtp.code()); // "564214"
  }
}